css-align-3/resources/alignment-parsing-utils.js

```diff
@@ -3,51 +3,41 @@ var contentPositionValues = [ "start", "end", "left", "right", "center", "flex-s
 var distributionValues = [ "stretch", "space-around", "space-between", "space-evenly"];
 var baselineValues = [ "baseline", "first baseline", "last baseline"];
 
-function checkPlaceContent(alignValue, justifyValue = "")
+function checkPlaceShorhand(shorthand, alignValue, justifyValue)
 {
-    checkPlaceShorhand("place-content", "align-content", "justify-content", alignValue, justifyValue);
-}
-
-function checkPlaceItems(alignValue, justifyValue = "")
-{
-    checkPlaceShorhand("place-items", "align-items", "justify-items", alignValue, justifyValue);
-}
-
-function checkPlaceSelf(alignValue, justifyValue = "")
-{
-    checkPlaceShorhand("place-self", "align-self", "justify-self", alignValue, justifyValue);
+    var div = document.createElement("div");
+    var value = (alignValue + " " + justifyValue).trim();
+    div.style[shorthand] = value;
+    document.body.appendChild(div);
+    var specifiedValue = alignValue;
+    if (alignValue !== justifyValue)
+        specifiedValue = value;
+    var resolvedValue = getComputedStyle(div).getPropertyValue(shorthand);
+    assert_equals(div.style[shorthand], specifiedValue, shorthand + " specified value");
+    assert_true(resolvedValue === value || resolvedValue === "", shorthand + " resolved value");
```

Comment:
Nit: Maybe it's worth add a FIXME and a comment pointing to https://github.com/w3c/csswg-drafts/issues/1041.